pilot/pkg/xds/ads.go

```diff
@@ -713,6 +713,14 @@ func AdsPushAll(s *DiscoveryServer) {
 // Primary code path is from v1 discoveryService.clearCache(), which is added as a handler
 // to the model ConfigStorageCache and Controller.
 func (s *DiscoveryServer) AdsPushAll(version string, req *model.PushRequest) {
+	// If we don't know what updated, cannot safely cache. Clear the whole cache
+	if len(req.ConfigsUpdated) == 0 {
+		s.cache.ClearAll()
+	} else {
+		// Otherwise, just clear the updated configs
+		s.cache.ClearHostnames(model.ConfigsOfKind(req.ConfigsUpdated, gvk.ServiceEntry))
+		s.cache.ClearDestinationRules(model.ConfigsOfKind(req.ConfigsUpdated, gvk.DestinationRule))
+	}
```

Comment:
Can we move above logic to `DiscoveryServer.pushConnection` in order to reduce calls times?

Reply:
This is called 1x per debounce. pushConnection is called Nx per debounce, were N is number of connections. Its also not accurate as we would clear out updates from previous proxies in a single push